evap/static/ts/src/notebook.ts

```diff
@@ -0,0 +1,89 @@
+declare const bootstrap: typeof import("bootstrap");
+import { unwrap, assert } from "./utils.js";
+
+const NOTEBOOK_LOCALSTORAGE_KEY = "evap_notebook_open";
+const COLLAPSE_TOGGLE_BUTTON_ID = "notebookButton";
+const WEBSITE_CONTENT_ID = "evapContent";
+const NOTEBOOK_FORM_ID = "notebook-form";
+
+class NotebookFormLogic {
+    private readonly notebook: HTMLFormElement;
+    // add a class constant
+    private readonly UPDATE_COOLDOWN = 2000;
+
+    constructor(notebookFormId: string) {
+        this.notebook = unwrap(document.getElementById(notebookFormId)) as HTMLFormElement;
+    }
+
+    private onSubmit = (event: SubmitEvent): void => {
+        event.preventDefault();
+
+        const submitter = unwrap(event.submitter) as HTMLButtonElement;
+        submitter.disabled = true;
+        this.notebook.setAttribute("data-state", "sending");
+
+        fetch(this.notebook.action, {
+            body: new FormData(this.notebook),
+            method: "POST",
+        })
+            .then(response => {
+                assert(response.ok);
+                this.notebook.setAttribute("data-state", "successful");
+                setTimeout(() => {
+                    this.notebook.setAttribute("data-state", "ready");
+                    submitter.disabled = false;
+                }, this.UPDATE_COOLDOWN);
+            })
+            .catch(() => {
+                this.notebook.setAttribute("data-state", "error");
+                submitter.disabled = false;
+                alert(submitter.dataset.errormessage);
+            });
+    };
+
+    public attach = (): void => {
+        this.notebook.addEventListener("submit", this.onSubmit);
+    };
+}
+
+export class NotebookLogic {
+    private readonly notebook_card: HTMLElement;
+    private readonly evap_content: HTMLElement;
+    private form_logic: NotebookFormLogic;
+    private readonly localStorageKey: string;
+
+    constructor(notebookId: string) {
+        this.notebook_card = unwrap(document.getElementById(notebookId));
+        this.form_logic = new NotebookFormLogic(NOTEBOOK_FORM_ID);
+        this.evap_content = unwrap(document.getElementById(WEBSITE_CONTENT_ID));
+        this.localStorageKey = NOTEBOOK_LOCALSTORAGE_KEY + "_" + this.notebook_card.dataset.notebookId;
```

Comment:
Inconsistent variable naming scheme - camelCase vs snake_case

(also notebookId)

Reply:
note: in js / ts we use camel case

Reply:
renamed to camelcase

Reply:
`notebook_card` still uses snake_case.